index.coffee

```diff
@@ -132,6 +131,8 @@ class ProgressReporter
 		return (evt) =>
 			try
 				{ id, status } = evt
+				id ?= ''
+				status ?= ''
```

Comment:
`status` could be undefined, which caused the _"Cannot read property 'startsWith' of undefined"_ error.

Reply:
You can move these into the dereferencing as:
```
{ id = '', status = '' } = evt

Reply:
Nice! I didn't know (or had forgotten) this was possible. However, this would not cover the case where id or status were `null` in the `evt` object, e.g.:
```
{ id = '', status = '' } = { id = null, status = null }
```
I don't know if it is possible for `status` to be null in `evt`, but I would err in the side of caution (especially as this same PR will start failing the builds on any progress errors), so I will keep the original version of the code.

Reply:
Actually due to this being coffee-script 1 it'll handle both `null` and `undefined`, it's coffee-script 2 and js/ts that only do it for `undefined`

Reply:
Thanks, good to realise that my VS Code CoffeeScript preview is using a different version of the coffeescript compiler than `package.json`! Now the new argument for not changing the code is that I wouldn't want it to break when/if someone updated the coffeescript version in `package.json`. :-)